src/MoBi.Presentation/Tasks/OutputSelectionsTask.cs

```diff
@@ -0,0 +1,150 @@
+         if(pathFromSimulations == null)
+            return;
+
+         if (!validateAndWarnNewPath(pathFromSimulations, simulationSettings))
+            return;
+
+         simulationSettings.OutputSelections.AddOutput(new QuantitySelection(pathFromSimulations, QuantityType.Undefined));
+         projectHasChanged();
+      }
+
+      private void showWarningMessage(ObjectPath pathFromSimulations)
+      {
+         _dialogCreator.MessageBoxInfo(AppConstants.Validation.ThePathIsAlreadySelectedAsAnOutput(pathFromSimulations));
+      }
+
+      private bool alreadyContainsPath(ObjectPath pathFromSimulations, SimulationSettings simulationSettings)
+      {
+         return simulationSettings.OutputSelections.Any(selection => Equals(selection.Path, pathFromSimulations.ToPathString()));
+      }
+
+      private void projectHasChanged()
+      {
+         _context.Project.HasChanged = true;
+      }
+
+      public void EditOutputSelection(SimulationSettings simulationSettings, QuantitySelection selection)
+      {
+         var newPath = selectPathFromSimulationsFor(AppConstants.Captions.ChangeDefaultCurveForNewSimulations, selection);
+
+         if (noChangesRequired(selection, newPath))
+            return;
+
+         if(validateAndUpdate(selection, newPath, simulationSettings))
+            projectHasChanged();
+      }
+
+      private static bool noChangesRequired(QuantitySelection selection, ObjectPath newPath)
+      {
+         return newPath == null || Equals(selection.Path, newPath.ToPathString());
+      }
+
+      private ObjectPath selectPathFromSimulationsFor(string caption, QuantitySelection preSelectedQuantity = null)
+      {
+         using (var modalPresenter = _applicationController.Start<IModalPresenter>())
+         {
+            modalPresenter.Text = caption;
+            var selectPathPresenter = _applicationController.Start<IHierarchicalQuantitySelectionPresenter>();
+            modalPresenter.Encapsulate(selectPathPresenter);
+            selectPathPresenter.SelectPathFrom(_simulationRepository.All().ToList());
+
+            if(preSelectedQuantity != null)
+               selectPathPresenter.SelectQuantityFromPath(new ObjectPath(preSelectedQuantity.Path.ToPathArray()));
+
+            if (modalPresenter.Show())
+               return selectPathPresenter.SelectedPath;
+         }
+
+         return null;
+      }
+
+      private bool validateAndWarnNewPath(ObjectPath selectedPath, SimulationSettings simulationSettings)
+      {
+         if (!alreadyContainsPath(selectedPath, simulationSettings)) 
+            return true;
+         
+         showWarningMessage(selectedPath);
+         return false;
+      }
+
+      public void UpdateOutputSelection(SimulationSettings simulationSettings, QuantitySelection selection, string newPath)
+      {
+         if (noChangesRequired(selection, new ObjectPath(newPath.ToPathArray())))
```

Comment:
maybe a private method to create and objectPath from a path as string (don't we have that already? somewhere)

Reply:
We have ObjectPathFactory, but you still have to use ```xxx.ToPathArray```

Should we add that in Core? Something like?

```
// something new
      public FormulaUsablePath CreateFormulaUsablePathFrom(string path)
      {
         return CreateFormulaUsablePathFrom(path.ToPathArray());
      }

// this is already there
      public FormulaUsablePath CreateFormulaUsablePathFrom(params string[] entries)
      {
         return CreateFormulaUsablePathFrom(entries.ToList());
      }
```

Or just an extension method in MoBi?